dev/com.ibm.ws.opentracing.1.2/src/com/ibm/ws/opentracing/OpentracingContainerFilter.java

```diff
@@ -86,14 +86,27 @@ public void filter(ContainerRequestContext incomingRequestContext) throws IOExce
             }
         }
 
+        String buildSpanName = null;
+        if (helper != null) {
+            buildSpanName = helper.getBuildSpanName(incomingRequestContext, resourceInfo);
+            if (buildSpanName == null) {
+                if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                    Tr.debug(tc, methodName + " skipping not traced method");
+                }
+                incomingRequestContext.setProperty(SERVER_SPAN_SKIPPED_ID, true);
+                return;
+            }
+        }
+
         URI incomingUri = incomingRequestContext.getUriInfo().getRequestUri();
         String incomingPath = incomingRequestContext.getUriInfo().getPath();
         if (!incomingPath.startsWith("/")) {
             incomingPath = "/" + incomingPath;
         }
 
-        String incomingURL = incomingUri.toURL().toString();
+        String incomingURL = null;
         if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+            incomingURL = incomingUri.toURL().toString();
             Tr.debug(tc, methodName + " incomingURL", incomingURL);
         }
 
```

Comment:
Either you can do the same thing you did with incomingURL with:

```
        SpanContext priorOutgoingContext = tracer.extract(Format.Builtin.HTTP_HEADERS,
                                                          new MultivaluedMapToTextMap(incomingRequestContext.getHeaders()));

        if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
            Tr.debug(tc, methodName + " priorContext", priorOutgoingContext);
        }
```
or can move it into the `if (process)`  conditional if @fmhwong is ok with that.

Reply:
The same would go for 1.3 and 2.0 as well.

Reply:
@fmhwong - can I move that to the `if (process)`or does the trace need to be before it?